src/ray/gcs/gcs_server/test/gcs_actor_scheduler_test.cc

```diff
@@ -15,349 +15,350 @@
-
-  void RunAndWait() {
-    Run();
-    Wait();
-  }
-
-  void CancelOnNode(ClientID node_id) {
-    auto actor_ids = gcs_actor_scheduler_->CancelOnNode(node_id);
-    canceled_actors_.insert(canceled_actors_.end(), actor_ids.begin(), actor_ids.end());
-    TryNotify();
-  }
-
-  void CancelOnWorker(ClientID node_id, WorkerID worker_id) {
-    auto actor_id = gcs_actor_scheduler_->CancelOnWorker(node_id, worker_id);
-    if (!actor_id.IsNil()) {
-      canceled_actors_.emplace_back(actor_id);
-      TryNotify();
-    }
-  }
-
-  void TryNotify() {
-    if (canceled_actors_.size() + succeed_actors_.size() + failed_actors_.size() ==
-        actor_count_) {
-      promise_.set_value(true);
-    }
-  }
-
-  bool ElementsEqual(const std::vector<std::shared_ptr<gcs::GcsActor>> &expected,
-                     const std::vector<std::shared_ptr<gcs::GcsActor>> &target) const {
-    if (expected.size() != target.size()) {
-      return false;
-    }
-    for (auto &actor : expected) {
-      if (std::find(target.begin(), target.end(), actor) == target.end()) {
-        return false;
-      }
-    }
-    return true;
+        /*lease_client_factory=*/
+        [this](const rpc::Address &address) { return raylet_client_; },
+        /*client_factory=*/
+        [this](const rpc::Address &address) { return worker_client_; });
   }
 
  protected:
-  const size_t actor_count_ = 100;
-  size_t node_count_ = 1;
-
   boost::asio::io_service io_service_;
+  Mocker::MockedActorInfoAccessor actor_info_accessor_;
   Mocker::MockedNodeInfoAccessor node_info_accessor_;
   Mocker::MockedErrorInfoAccessor error_info_accessor_;
-  Mocker::MockedActorInfoAccessor actor_info_accessor_;
-  std::shared_ptr<gcs::GcsNodeManager> gcs_node_manager_;
-  std::shared_ptr<gcs::GcsActorScheduler> gcs_actor_scheduler_;
-  std::unique_ptr<std::thread> io_thread_;
-  // Timeout waiting for gcs server reply, default is 5s
-  const uint64_t timeout_ms_ = 2000;
-
-  std::vector<std::shared_ptr<gcs::GcsActor>> to_be_scheduled_actors_;
-  std::vector<std::shared_ptr<gcs::GcsActor>> succeed_actors_;
-  std::vector<std::shared_ptr<gcs::GcsActor>> failed_actors_;
-  std::vector<ActorID> canceled_actors_;
-  std::promise<bool> promise_;
-};
 
-class ZeroNodeTest : public GcsActorSchedulerTest {
- public:
-  ZeroNodeTest() : GcsActorSchedulerTest(0) {}
+  std::shared_ptr<Mocker::MockRayletClient> raylet_client_;
+  std::shared_ptr<Mocker::MockWorkerClient> worker_client_;
+  std::shared_ptr<gcs::GcsNodeManager> gcs_node_manager_;
+  std::shared_ptr<Mocker::MockedGcsActorScheduler> gcs_actor_scheduler_;
+  std::vector<std::shared_ptr<gcs::GcsActor>> success_actors_;
+  std::vector<std::shared_ptr<gcs::GcsActor>> failure_actors_;
 };
 
-class OneNodeTest : public GcsActorSchedulerTest {
- public:
-  OneNodeTest() : GcsActorSchedulerTest(1) {}
-};
+TEST_F(GcsActorSchedulerTest, TestScheduleFailedWithZeroNode) {
+  ASSERT_EQ(0, gcs_node_manager_->GetAllAliveNodes().size());
```

Comment:
Usually the actual value goes first and the expected value goes second. Not a big deal, but it makes the output nicer if the values do not match.